skytap/resource_skytap_vm_test.go

```diff
@@ -527,6 +421,39 @@ func TestAccCasandra(t *testing.T) {
 	})
 }
 
+func testAccCheckSkytapExternalPorts(t *testing.T, vmName string, count string) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rsVM, err := getResource(s, vmName)
+		if err != nil {
+			return err
+		}
+		assert.Equal(t, count, rsVM.Primary.Attributes["service_ports.%"], "empty map entry")
+		assert.Equal(t, count, rsVM.Primary.Attributes["service_ips.%"], "empty map entry")
+
+		return nil
+	}
+}
+
+func setupEnvironmentWithKeys(templateKey string, templateIDFallback string, vmKey string, vmIDFallback string) (string, string, string) {
```

Comment:
not sure if the setup method helps much, consider to user named returns to simplify this method

Reply:
It gives some flexibility. I suggest we keep it as is.